tests/runtime/test_caikit_health_probe.py

```diff
@@ -85,54 +71,101 @@ def temp_probe_config(*args, **kwargs):
+        ProbeTestConfig(TlsMode.INSECURE, ServerMode.BOTH),
+        # TLS
+        ProbeTestConfig(TlsMode.TLS, ServerMode.HTTP),
+        ProbeTestConfig(TlsMode.TLS, ServerMode.GRPC),
+        ProbeTestConfig(TlsMode.TLS, ServerMode.BOTH),
+        ProbeTestConfig(TlsMode.TLS, ServerMode.BOTH, inline=True),
+        ProbeTestConfig(TlsMode.TLS, ServerMode.BOTH, localhost_in_cert=False),
+        # mTLS
+        ProbeTestConfig(TlsMode.MTLS, ServerMode.HTTP),
+        ProbeTestConfig(TlsMode.MTLS, ServerMode.GRPC),
+        ProbeTestConfig(TlsMode.MTLS, ServerMode.BOTH),
+        ProbeTestConfig(TlsMode.MTLS, ServerMode.BOTH, inline=True),
+        ProbeTestConfig(TlsMode.MTLS, ServerMode.BOTH, localhost_in_cert=False),
+        ProbeTestConfig(TlsMode.MTLS, ServerMode.BOTH, common_client_ca=False),
+        # Invalid configs that never pass
+        ProbeTestConfig(
+            TlsMode.TLS,
+            ServerMode.GRPC,
+            localhost_in_cert=False,
+            unix_socket=False,
+            should_become_healthy=False,
+        ),
+        ProbeTestConfig(
+            TlsMode.TLS,
+            ServerMode.BOTH,
+            localhost_in_cert=False,
+            unix_socket=False,
+            should_become_healthy=False,
+        ),
+        ProbeTestConfig(
+            TlsMode.MTLS,
+            ServerMode.GRPC,
+            localhost_in_cert=False,
+            unix_socket=False,
+            should_become_healthy=False,
+        ),
     ],
 )
-@pytest.mark.parametrize("server_mode", ServerMode.__members__.values())
-def test_health_probe(test_config: Tuple[TlsMode, bool], server_mode: ServerMode):
+def test_health_probe(test_config: ProbeTestConfig):
     """Test all of the different ways that the servers could be running"""
-    tls_mode, inline = test_config
-    with alog.ContextLog(
-        log.info,
-        "---LOG CONFIG: tls: %s, inline: %s, server: %s---",
-        tls_mode.name,
-        inline,
-        server_mode.name,
-    ):
+    with alog.ContextLog(log.info, "---LOG CONFIG: %s---", test_config):
         # Get ports for both servers
         http_port = tls_test_tools.open_port()
         grpc_port = tls_test_tools.open_port()
 
         # Set up SAN lists if not putting "localhost" in
         server_sans, client_sans = None, None
-        if "NO_LOCALHOST" in tls_mode.name:
+        if not test_config.localhost_in_cert:
             server_sans = ["foo.bar"]
             client_sans = ["baz.bat"]
 
         # Set up tls values if needed
-        tls = tls_mode.name.startswith("TLS")
-        mtls = tls_mode.name.startswith("MTLS")
         with generate_tls_configs(
             port=http_port,
-            tls=tls,
-            mtls=mtls,
-            inline=inline,
-            separate_client_ca=tls_mode == TlsMode.MTLS_SEPARATE_CLIENT_CA,
+            tls=test_config.tls_mode == TlsMode.TLS,
+            mtls=test_config.tls_mode == TlsMode.MTLS,
```

Comment:
nice, this is way better than name parsing

Reply:
Yeah, that was getting out of hand